### ISSUE TITLE
Crash with mismatched quotes in a local-exec command

### ISSUE THREAD

**Issue by ericpromislow** [hashicorp/terraform#4589]
I forgot to escape an inner double-quote and terraform crashed.
[crash.zip](https://github.com/hashicorp/terraform/files/83403/crash.zip)

Code:

    ...
    provisioner "local-exec" {
      command = "echo '"${var.password}"'
    }

You get the crash even if you put the double-quote at the end of the command string -- the problem is with those unescaped inner dquot's.

**Comment by mitchellh** [comment]
Hey @ericpromislow, I ran a test case with this config:

```
resource "test_instance" "test" {
  provisioner "local-exec" {
    command = "echo '"${var.password}"'
  }
}
```

And got this error:

```
Error parsing test-fixtures/mismatched_quote.tf: At 3:40: literal not terminated
```

Which seems nice! Maybe this was fixed? Or can you please share a config that reproduces this?

Thanks!

**Comment by ericpromislow** [comment]
Looks fixed in v0.6.9 - I get the error message and no crash. This is ready to close.

**Comment by catsby** [comment]
Thanks for following up @ericpromislow !
